src/main/java/mate/academy/internetshop/dao/jdbc/OrderDaoJdbcImpl.java

```diff
@@ -0,0 +1,163 @@
+            statement.setLong(1, userId);
+            ResultSet resultSet = statement.executeQuery();
+            List<Order> orders = new ArrayList<>();
+            while (resultSet.next()) {
+                orders.add(getOrderFromResultSet(resultSet).get());
+            }
+            return orders;
+        } catch (SQLException exception) {
+            throw new DataProcessingException(exception.getMessage(), exception);
+        }
+
+    }
+
+    private List<Product> getProductsFromOrder(Long id) throws SQLException {
+        String selectQuery = "SELECT * "
+                + "FROM products JOIN orders_products "
+                + "ON orders_products.product_id = products.id "
+                + "WHERE order_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(selectQuery);
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            List<Product> products = new ArrayList<>();
+            while (resultSet.next()) {
+                Product product = new Product(resultSet.getString("name"),
+                        resultSet.getBigDecimal("price"));
+                product.setId(resultSet.getLong("id"));
+                products.add(product);
+            }
+            return products;
+        }
+    }
+
+    private void addProductsToOrder(Order order) {
+        String insertQuery = "INSERT INTO orders_products (order_id, product_id)"
+                + " VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            for (Product product : order.getProducts()) {
+                PreparedStatement statement = connection.prepareStatement(insertQuery);
+                statement.setLong(1, order.getId());
+                statement.setLong(2, product.getId());
+                statement.executeUpdate();
+            }
+        } catch (SQLException exception) {
+            throw new DataProcessingException(exception.getMessage(), exception);
+        }
+    }
+
+    private void deleteProductsFromOrder(Long id) {
+        String deleteQuery = "DELETE FROM orders_products WHERE order_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(deleteQuery);
+            statement.setLong(1, id);
+            statement.executeUpdate();
+        } catch (SQLException exception) {
+            throw new DataProcessingException(exception.getMessage(), exception);
+        }
+    }
+
+    private Optional<Order> getOrderFromResultSet(ResultSet resultSet) throws SQLException {
+        Order order = new Order(getProductsFromOrder(resultSet.getLong("order_id")),
+                resultSet.getLong("user_id"));
+        order.setId(resultSet.getLong("order_id"));
+        return Optional.of(order);
```

Comment:
should we use Optional in private method which you are using only inside this dao